src/data/markdown/docs/05 Examples/01 Examples/24 distribute-workloads.md

```diff
@@ -0,0 +1,229 @@
+With the `exec` object, you can scope logic to a specific instance, scenario, or across all VUs.
+
+For example, this statement assigns behavior to the first 25 VUs in a test.
+
+```bash
+if (exec.vu.idInTest <= 25) {
+   //do something;
+  }
+```
+
+
+For more flexibility, you can use modulo expressions distribute VUs according to percentages.
+For example, the following script distributes logic according to different user profiles:
+- 40 percent of users check the news.
+- 60 percent play a coinflip game.
+   - Half bet `heads` and half bet `tails`.
+
+<CodeGroup labels={["behavior-based-on-exec-context.js"]} lineNumbers={[true]} showCopyButton={[true]}>
+
+```javascript
+import http from "k6/http";
+import exec from "k6/execution";
+
+export const options = {
+  scenarios: {
+    quickRamp: {
+      executor: "ramping-arrival-rate",
+      startRate: 0,
+      timeUnit: "1s",
+      preAllocatedVUs: 100,
+      stages: [
+        { target: 10, duration: "10s" },
+        { target: 10, duration: "15s" },
+        { target: 0, duration: "5s" },
+      ],
+    },
+  },
+};
+
+export default function () {
+  if (exec.vu.idInTest % 10 < 4) {
+    // 0-3 range, read the news
+    http.get("http://test.k6.io/news");
+  }
+  else if (exec.vu.idInTest % 10 < 7) {
+      // 4-6 range, bet heads
+    http.get("http://test.k6.io/flip_coin.php?bet=heads");
+  } else {
+    // 7-9 range, bet tails"
+    http.get("http://test.k6.io/flip_coin.php?bet=tails");
+  }
+}
+
+```
+
+</CodeGroup>
+
+## Random behavior
+
+To add a degree of random behavior, consider one of the randomizing functions from the [k6 utils](https://k6.io/docs/javascript-api/jslib/utils/).
+
+For example, this script randomly assigns one behavior to happen one third of the time, and another behavior to happen all other times.
+
+<CodeGroup labels={["random-behavior.js"]} lineNumbers={[]} showCopyButton={[true]}>
+
+```javascript
+import { sleep } from "k6";
+import { randomIntBetween } from "https://jslib.k6.io/k6-utils/1.2.0/index.js";
+
+export default function () {
+  if (randomIntBetween(1, 3) % 3 > 1) {
+    console.log("1 in 3 times");
+  } else {
+    console.log("2 in 3 times");
+  }
+}
+
+```
+
+</CodeGroup>
+
+For a more sophisticated example of randomizing, read this [forum post](https://community.k6.io/t/how-to-distribute-vus-across-different-scenarios-with-k6/49/17).
+
+## Inspect results by behavior
+
+If you want to inspect the results for a certain behavior, you have the following options:
+
+- [Create a custom metric](/using-k6/metrics/create-custom-metrics).
+  These would also appear in the end-of-test summary.
+- Use [Tags and groups](/using-k6/tags-and-groups)
+
+
+To view granular results for a specific scenario, you can filter results by the built in scenario [tag](/using-k6/tags-and-groups).
+For example, this command uses `jq` to filter only data points for the  scenario with the name "news".
```

Comment:
The `jq` instruction seems out of context.

Reply:
I think either delete it entirely or move it to the end of the scenarios section.